### PR TITLE
feat: Publish uikit

### PR DESCRIPTION
**Pull request by chef-jojo**
### Summary
🗑️🔄🎨

This pull request refactors the codebase to use `@pancakeswap/uikit` instead of `@pancakeswap/ui` and `styled-components` for UI components and styling. It updates the imports and removes the unused dependencies in various files.

> _`@pancakeswap/ui`_
> _Gone with the autumn wind, now_
> _`@pancakeswap/uikit`_

### Walkthrough
*  Remove unused dependency `@pancakeswap/ui` from `config.json` file ([link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-ed5b41335ff968c5cfd8035d4b8f8c8c0538b6746182d522adb0312eb9137fc5L12))
*  Replace imports of `@pancakeswap/ui` with `@pancakeswap/uikit` in various components that use `AtomBox` ([link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-8f38487f996466b32431174898b0ce16c25db23ad860e61eaccffa2fc58d6e59L1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-39cb98252ae39aa483e099b8c1ec9d6e90480e914df34550a5ec22c465a5d60eL2-R6))
*  Replace imports of `styled-components` with `@pancakeswap/uikit` in various components that use `styled`, `css` or `keyframes` ([link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-271d8ca2c32247fd2309fe8bda858c0ddea004370a974abd5273df841e42796dL3-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-d68130d4a406d32394548987c21ffa580a7a85b1c6d6b6d30376461d4937aabcL9-R13), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-c833f928225e8d1bd05b8cf3e82a276f8c95040ad243c9e315bd3984d7398f9eL1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-e7713e0b17198c1f827b60cba14a6145cf7e09c219a8270daf717c52d611bacfL4-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-bcd987a55beb759352c064ec566201489281f100fa7e6583933bce5339554759L1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-8b765322f2ea8353d33299240988ee1b012663574645d61810dbc0af633ff97fL4-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-1bcc706e1f73c1fdf441e5e53831aee7a9922a8bf14f7683eac2ba2cc933639eL3-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-cedd50245481321c9908a963877cf4e2dff2d843509a02d1f1f6426c2db5d15bL11-R11), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-cf1bb104c1dd1e9c54355d80d359f2dfeaa212383ad3027756510f63183483b5L1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-afeee2ff7a2b0a1bd1222f363b04453dfdee7562ebcf294448bced5331718e7eL2-R2), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-9ee711fe5bdd70f4933930da0842f6ac5a59fa752d84fdf15e1ad8ae2a546d3bL17-R17), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-e0a5accbd3cd0970d5c03017ff1c08690c2847c1bd01aec6294836c61b9cfe0dL23-R23), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-9a571e024a13388e61d76b6a76009d42b488314dffa0efd772ca68e73d3cad4eL3-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-9e80d574803a670402652154dd1106cff393506eeda94373e2770538bb49e84aL1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-7b2804c528fd7ed307fb068f6960ecbe58da82913657bdca32b1e41f50fbef68L3-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-b6d1212e922ae2f178f0d70bca3d369ee97f9c4ac810421e614125910e278e21L19-R19), [link](https://github.com/pancakeswap/pancake-frontend/pull/7723/files?diff=unified&w=0#diff-c07a30883c90aae88f6b9184bf7c7d977646c64b214f7b500cb0bc493daa7b65L7-R7))

### CODE REVIEW

**Comment by chef-jojo** [comment]
removed prefix vender by default 
https://styled-components.com/docs/faqs#vendor-prefixes-are-omitted-by-default

**Comment by chef-jojo** [comment]
@SocketSecurity ignore-all

File: packages/uikit/package.json [hunk 1]
```diff
@@ -83,50 +108,57 @@
...
-    "@pancakeswap/ui": "workspace:*",
+    "@pancakeswap/farms": "workspace:*",
+    "@pancakeswap/hooks": "workspace:*",
+    "@pancakeswap/localization": "workspace:*",
+    "@pancakeswap/pools": "workspace:*",
+    "@pancakeswap/sdk": "workspace:*",
+    "@pancakeswap/swap-sdk-core": "workspace:*",
+    "@pancakeswap/token-lists": "workspace:*",
+    "@pancakeswap/tokens": "workspace:*",
+    "@pancakeswap/utils": "workspace:*",
```
Comment (chefjackson) [inline packages/uikit/package.json:145]: let's move @pancakeswap/utils to devDependencies as it's not published. And probably need to bundle it into the uikiit? For other pkgs we add it into `noExternal`

Comment (chef-jojo) [reply packages/uikit/package.json]: seems like vite detect it as no external
I was on purpose keeping utils private to enable inline bundling

**Review by chefjackson — APPROVED** [review]
Great job! Cheers!

**Pull request merged**